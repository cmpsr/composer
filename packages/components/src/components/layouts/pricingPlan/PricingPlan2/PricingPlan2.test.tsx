import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { PricingPlan2, PRICING_PLAN_DEFAULT_TEST_ID } from '.';

describe('PricingPlan2', () => {
  const title = 'test';
  const price = '10';
  const items = ['test 1', 'test 2', 'test 3', 'test 4'];
  const button = <a>test</a>;

  it('should match snapshot', () => {
    const pricingPlan = renderer.create(
      <PricingPlan2 title={title} price={price}>
        {{ listItems: items, button: button }}
      </PricingPlan2>,
    );
    const tree = pricingPlan.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render component', () => {
    const { getByTestId } = render(
      <PricingPlan2 title={title} price={price} />,
    );
    const pricingPlan = getByTestId(PRICING_PLAN_DEFAULT_TEST_ID);
    expect(pricingPlan).toBeInTheDocument();
  });
  it('should render className', () => {
    const { getByTestId } = render(
      <PricingPlan2 title={title} price={price} className="foo" />,
    );
    const pricingPlan = getByTestId(PRICING_PLAN_DEFAULT_TEST_ID);
    expect(pricingPlan.classList).toContain('foo');
  });
  it('should render title', () => {
    const { getByText } = render(<PricingPlan2 title={title} price={price} />);
    const pricingPlan = getByText('test');
    expect(pricingPlan).toBeInTheDocument();
  });
  it('should render price', () => {
    const { getByText } = render(<PricingPlan2 title={title} price={price} />);
    const pricingPlan = getByText('$10');
    expect(pricingPlan).toBeInTheDocument();
  });
  it('should render children list', () => {
    const { getByText } = render(
      <PricingPlan2 title={title} price={price}>
        {{ listItems: items }}
      </PricingPlan2>,
    );
    const pricingPlan = getByText('test 1');
    expect(pricingPlan).toBeInTheDocument();
  });
  it('should render children button', () => {
    const childrenTestId = 'Children-TestId';
    const link = <a data-testid={childrenTestId}>test</a>;
    const { getByTestId } = render(
      <PricingPlan2 title={title} price={price}>
        {{ listItems: items, button: link }}
      </PricingPlan2>,
    );
    const pricingPlan = getByTestId(childrenTestId);
    expect(pricingPlan).toBeInTheDocument();
  });
});
