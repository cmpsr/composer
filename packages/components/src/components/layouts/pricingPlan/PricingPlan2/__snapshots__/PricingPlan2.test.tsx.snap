// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricingPlan2 should match snapshot 1`] = `
<div
  className="rounded shadow-lg px-6 py-8"
  data-testid="pricingPlan2"
>
  <div
    className="flex justify-between items-center"
  >
    <h5
      className="typo-headline-5 font-bold"
      data-testid="typography"
    >
      test
    </h5>
    <div
      className="flex flex-row items-baseline"
    >
      <span
        className="typo-headline-5 font-bold"
        data-testid="typography"
      >
        $
        10
      </span>
      <span
        className="typo-body ml-2"
        data-testid="typography"
      >
        monthly
      </span>
    </div>
  </div>
  <div
    className="mt-6 mb-10"
  >
    <ul>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 1
        </span>
      </li>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 2
        </span>
      </li>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 3
        </span>
      </li>
      <li
        className="mb-3"
      >
        <svg
          className="inline-block mr-5"
          data-testid="icon"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <span
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          test 4
        </span>
      </li>
    </ul>
  </div>
  <a>
    test
  </a>
</div>
`;
